src/App/App.js

```diff
@@ -10,25 +11,43 @@ const beValues = [
   {
     id: '0',
     caption: 'Card one header',
-    content: loremIpsum
+    content: loremIpsum,
+    selectable: true,
+    selected: true
   }, {
     id: '1',
     caption: 'Card two header',
-    content: loremIpsum
+    content: loremIpsum,
+    selectable: true,
+    selected: true
   }, {
     id: '2',
     caption: 'Card three header',
-    content: loremIpsum
+    content: loremIpsum,
+    selectable: true,
+    selected: false
   }
 ];
 
 
 function App() {
-  const cards = beValues.map(one =>
+  const [cardsState, setCardsState] = useState(beValues);
+
+  const cardsRendered = cardsState.map((one, index) =>
     <Card
       key={one.id}
       caption={one.caption}
       content={one.content}
+      checkbox={one.selectable}
+      selected={one.selected}
+      onSelect={() => {
+        const newCardsState = [...cardsState];
+        newCardsState[index] = {
+          ...one,
+          selected: !one.selected
+        }
+        setCardsState(newCardsState);
```

Comment:
Can you please replace you construction by "() => setCardsState(...)"?
Also it would be better to orient by card id (not using index).